src/Nomad/ModifiableImagesCollection.cs

```diff
@@ -81,35 +82,37 @@ public async Task RemoveImageAsync(IFile imageFile, CancellationToken cancellati
     /// <remarks>
     /// This method will call <see cref="ReadOnlyImagesCollection.GetAsync(string, CancellationToken)"/> and create a new instance to pass to the event handlers.
     /// <para/>
-    /// If already have a resolved instance of <see cref="Image"/>, you should call <see cref="ApplyEntryUpdateAsync(ValueUpdateEvent, Image, CancellationToken)"/> instead.
+    /// If already have a resolved instance of <see cref="Image"/>, you should call <see cref="ApplyEntryUpdateAsync(EventStreamEntry{DagCid}, ValueUpdateEvent, Image, CancellationToken)"/> instead.
     /// </remarks>
+    /// <param name="eventStreamEntry">The event stream entry to apply.</param>
     /// <param name="updateEvent">The update event to apply.</param>
     /// <param name="cancellationToken">A token that can be used to cancel the ongoing operation.</param>
-    public override async Task ApplyEntryUpdateAsync(ValueUpdateEvent updateEvent, CancellationToken cancellationToken)
+    public override async Task ApplyEntryUpdateAsync(EventStreamEntry<DagCid> eventStreamEntry, ValueUpdateEvent updateEvent, CancellationToken cancellationToken)
     {
         cancellationToken.ThrowIfCancellationRequested();
         
-        if (updateEvent.TargetId != Id)
+        if (eventStreamEntry.TargetId != Id)
             return;
         
         Guard.IsNotNull(updateEvent.Value);
         var (image, _) = await Client.ResolveDagCidAsync<Image>(updateEvent.Value.Value, nocache: !KuboOptions.UseCache, cancellationToken);
         
         Guard.IsNotNull(image);
-        await ApplyEntryUpdateAsync(updateEvent, image, cancellationToken);
+        await ApplyEntryUpdateAsync(eventStreamEntry, updateEvent, image, cancellationToken);
     }
 
     /// <summary>
     /// Applies an event stream update event and raises the relevant events.
     /// </summary>
+    /// <param name="eventStreamEntry">The event stream entry to apply.</param>
     /// <param name="updateEvent">The update event to apply.</param>
     /// <param name="image">The resolved image data for this event.</param>
     /// <param name="cancellationToken">A token that can be used to cancel the ongoing operation.</param>
-    public async Task ApplyEntryUpdateAsync(ValueUpdateEvent updateEvent, Image image, CancellationToken cancellationToken)
+    public async Task ApplyEntryUpdateAsync(EventStreamEntry<DagCid> eventStreamEntry, ValueUpdateEvent updateEvent, Image image, CancellationToken cancellationToken)
     {
         cancellationToken.ThrowIfCancellationRequested();
 
-        switch (updateEvent.EventId)
+        switch (eventStreamEntry.EventId)
         {
             case nameof(AddImageAsync):
             {
```

Comment:
shouldn't we add some error handling when getting the image?

Reply:
No, we rely on .NET exception throwing and don't need to do manual error handling. If the image isn't found, the consumer of this method should try/catch at their callsite, not here.